ReactJS/lesson_4/src/app/components/AddNewPost.js

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import '../styles/addNewPost.css'
+
+export default class AddNewPost extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      userName: '',
+      title: '',
+      body: '',
+      display: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  
+  handleChange(event) {
+    this.setState({
+      [ event.currentTarget.name ]: event.target.value
+    });
+  }
+  
+  handleSubmit(event) {
+    event.preventDefault();
+    if (this.state.userName.length && this.state.title.length && this.state.body.length) {
+      delete this.state.display;
```

Comment:
```suggestion
      const { display, ...post } = this.state;;
```
не мутируйте (изменяйте) состояния:)